pkg/teleworker/job.go

```diff
@@ -0,0 +1,186 @@
+package teleworker
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/gofrs/uuid"
+	api "github.com/spirifoxy/teleworker/internal/api/v1"
+	ls "github.com/spirifoxy/teleworker/internal/logstreamer"
+	cg "github.com/spirifoxy/teleworker/pkg/cgroup"
+)
+
+type Limits struct {
+	MemoryMB  int
+	CpuWeight int
+	IOWeight  int
+}
+
+// ToCgroupLimits formats limits in format acceptable as
+// cgroup parameters and return them as strings
+func (l *Limits) ToCgroupLimits() cg.Limits {
+	formatted := cg.Limits{}
+	if l.MemoryMB > 0 {
+		formatted[cg.MemLimit] = fmt.Sprintf("%dM", l.MemoryMB)
+	}
+
+	if l.CpuWeight > 0 {
+		formatted[cg.CpuShares] = strconv.Itoa(l.CpuWeight * 10)
+	}
+
+	if l.IOWeight > 0 {
+		formatted[cg.BlkioWeight] = strconv.Itoa(l.IOWeight * 10)
+	}
+
+	return formatted
+}
+
+func (l *Limits) ToFlags() []string {
+	flags := []string{}
+	if l.MemoryMB > 0 {
+		flags = append(flags, fmt.Sprintf("-memorymb=%d", l.MemoryMB))
+	}
+
+	if l.CpuWeight > 0 {
+		flags = append(flags, fmt.Sprintf("-cpuweight=%d", l.CpuWeight))
+	}
+
+	if l.IOWeight > 0 {
+		flags = append(flags, fmt.Sprintf("-ioweight=%d", l.IOWeight))
+	}
+	return flags
+}
+
+type JobState struct {
+	Status   api.JobStatus
+	ExitCode int
+	ExitErr  error
+	ExitedAt time.Time
+	Limits   *Limits
+}
+
+// StatusName returns user friendly name for JobStatus
+// See api.JobStatus_name for code to string mapping
+func (js *JobState) StatusName() string {
+	name, ok := api.JobStatus_name[int32(js.Status)]
+	if !ok {
+		// Fallback to the default UNKNOWN status,
+		// but should never happen in reality
+		return api.JobStatus_name[0]
+	}
+	return name
+}
```

Comment:
Does something like `js.Status.String()` work for this?

Reply:
yeah, this StatusName is actually not needed at all, thanks. Removed